### PR TITLE
feature/better db specific types

### PR DESCRIPTION
**Pull request by yankeeinlondon**
This PR's main intent is to improve DB support by:

1. Creating a separate function for each database to map that DB's type to a result
2. Added the "mssql" feature -- which **sqlx** supports and added all the conditional blocks you _should_ need for a quick PR. There is a PR needed to open up use though as the API surface is different. This is documented in the README.

In addition to the core focus above, two additional things can be found in this PR:

- **Tracing** - to aid upstream users of this library and help us work through any edge cases around type conversion (amongst other things), this library now publishes spans using Tokio's `tracing` crate. Note that there is a "dev dependency" on `tracing-subscriber` too as the intent is to bring some of this trace information into future tests. This dev-dep has zero cost to the library.
- **Graceful Exit** - while the `dev` branch now allows for explicit closing of the DB pools this still requires that userland handle this call; this PR maintains the ability to close() whenever needed but adds automatic DB closing when the app is shutting down

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -8,15 +8,25 @@ edition = "2021"
...
 
 [features]
 sqlite = ["sqlx/sqlite"]
 mysql = ["sqlx/mysql"]
 postgres = ["sqlx/postgres"]
+mssql = ["sqlx/mssql"]
+
+[profile.release]
+lto = true
+opt-level = 3
```
Comment (FabianLars) [inline Cargo.toml]:
```suggestion
```

cargo ignores profiles of dependencies so this has no effect.

File: example/vue-todos/package.json [hunk 2]
```diff
@@ -15,35 +16,35 @@
...
+    "@vueuse/core": "^9.4.0",
+    "pinia": "^2.0.23",
+    "tauri-plugin-sql-api": "workspace:0.1.0",
     "uuid": "^9.0.0",
-    "vue": "^3.2.40"
+    "vue": "^3.2.41"
   },
   "devDependencies": {
-    "@iconify/json": "^2.1.120",
+    "@iconify/json": "^2.1.129",
```
Comment (FabianLars) [inline example/vue-todos/package.json]: btw do we really need the iconify dependency? That always takes _ages_ for me to download

File: example/vue-todos/package.json [hunk 3]
```diff
@@ -15,35 +16,35 @@
...
   "dependencies": {
-    "@vueuse/components": "^9.3.0",
-    "@vueuse/core": "^9.3.0",
-    "pinia": "^2.0.23",
     "@tauri-apps/api": "^1.1.0",
-    "tauri-plugin-sql-api": "github:tauri-apps/tauri-plugin-sql#dev",
+    "@vueuse/components": "^9.4.0",
+    "@vueuse/core": "^9.4.0",
+    "pinia": "^2.0.23",
+    "tauri-plugin-sql-api": "workspace:0.1.0",
```
Comment (FabianLars) [inline example/vue-todos/package.json]:
```suggestion
    "tauri-plugin-sql-api": "../../",
```

**Review by FabianLars — CHANGES_REQUESTED** [review]
oof okay we need to start splitting up prs into multiple ones, this a bit much :D

Anyway...
Can we rename example to examples again? examples is the default for cargo projects (and i think the only accepted name), which would especially be good if we ever end up with a Rust api which could get a rust-only example. Also this would result in consistency across all repos.
I don't have hard feelings bout that tho.

**i didn't look at the deserialize.rs file yet**

File: example/vue-todos/package.json (hunk 2 above)
Comment (yankeeinlondon) [reply example/vue-todos/package.json]: i've never noticed it taking long on my side but my bandwidth is 1gbe; I should be able to replace this with a subset of icon packs. When developing it's nice just to have the freedom to use across icon libraries and this is just a dev-dep so no extra bits being shipped. Anyway, yeah let me make this smaller.

File: Cargo.toml (hunk 1 above)
Comment (yankeeinlondon) [reply Cargo.toml]: ahhh; removing now.

File: example/vue-todos/package.json (hunk 3 above)
Comment (yankeeinlondon) [reply example/vue-todos/package.json]: are you ok if I just make this a workspace? forgot to move that over from the other PR.

Comment (yankeeinlondon) [reply example/vue-todos/package.json]: Ok, I actually did have the workspace defined. Not sure why it doesn't work here but I can go back to the ugly way.

**Comment by yankeeinlondon** [comment]
> oof okay we need to start splitting up prs into multiple ones, this a bit much :D
> 
> Anyway... Can we rename example to examples again? examples is the default for cargo projects (and i think the only accepted name), which would especially be good if we ever end up with a Rust api which could get a rust-only example. Also this would result in consistency across all repos. I don't have hard feelings bout that tho.
> 
> **i didn't look at the deserialize.rs file yet**

I just copied the other PR's very mildly updated example code (updated deps etc). Nothing needs review in here.

WRT to `examples` vs `example` ... I don't care at all but thought it was more linguistically correct and actually (probably mistakenly) thought that that's what this repo had called it originally before I added my example.

**Pull request closed without merging**